ddht/v5_1/alexandria/network.py

```diff
@@ -0,0 +1,131 @@
+import logging
+from typing import AsyncContextManager, Optional, Type
+
+from async_service import Service
+from eth_enr import ENRManagerAPI
+from eth_typing import NodeID
+import trio
+
+from ddht.base_message import InboundMessage
+from ddht.endpoint import Endpoint
+from ddht.exceptions import DecodingError
+from ddht.subscription_manager import SubscriptionManager
+from ddht.v5_1.abc import NetworkAPI
+from ddht.v5_1.alexandria.abc import AlexandriaClientAPI, AlexandriaNetworkAPI
+from ddht.v5_1.alexandria.constants import ALEXANDRIA_PROTOCOL_ID
+from ddht.v5_1.alexandria.messages import (
+    PingMessage,
+    TAlexandriaMessage,
+    decode_message,
+)
+from ddht.v5_1.alexandria.payloads import PongPayload
+from ddht.v5_1.messages import TalkRequestMessage, TalkResponseMessage
+
+
+class AlexandriaNetwork(Service, AlexandriaNetworkAPI):
+    logger = logging.getLogger("ddht.Alexandria")
+
+    protocol_id = ALEXANDRIA_PROTOCOL_ID
+
+    _alexandria_client: AlexandriaClientAPI
+
+    def __init__(self, client: AlexandriaClientAPI) -> None:
+        self._alexandria_client = client
+        self.protocol_id = self._alexandria_client.protocol_id
```

Comment:
*nit* it feels a little weird to set the `protocol_id` both here and in the top-level of the class.

It seems like this class, at the moment, makes almost no assumptions about what the protocol looks like, so maybe it should accept a protocol_id from the client.

If in the future it'll know more about the Alexandria protocol it might be better for this line to be an assert, that the client is using the same protocol we're using.